SystemLog: проверка структуры и существования таблицы при каждом flush

kshvakov opened the pull request:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Сейчас `prepareTable` вызывается только один раз, проблема возникает если удалили таблицу системных логов, например для очистки старых данных или случайно. 
Если это произошло, то приходиться перезагружать сервер или пытаться найти структуру таблицы на другом сервере и создать её руками. Для избежания подобных проблем можно проверять таблицу логов каждый раз при `flush`.

kshvakov committed:
SystemLog: execute prepareTable()  on each flush

kshvakov committed:
Merge remote-tracking branch 'upstream/master' into system-log-prepare-table-on-each-flush

alexey-milovidov commented:
Излишнее логгирование:
```
LOG_DEBUG(log, "Will use existing table " << description << " for " + LogElement::name());
```

alexey-milovidov reviewed (changes requested):
.

kshvakov committed:
#2131 deleted excessive logging

alexey-milovidov commented:
А теперь недостаточное логгирование (потеряно логгирование при старте).

kshvakov committed:
Revert "#2131 deleted excessive logging"

This reverts commit d78a46d958cffeabf9543cb0e35e744c7e141bd6.

kshvakov committed:
Revert "SystemLog: execute prepareTable()  on each flush"

This reverts commit 7f60bb0a9b62c991510457efac3386dc573baafb.

kshvakov committed:
SystemLog: execute prepareTable() on each flush

kshvakov committed:
SystemLog: execute prepareTable()  on each flush

Merge remote-tracking branch 'upstream/master' into system-log-prepare-table-on-each-flush

#2131 deleted excessive logging

Revert "#2131 deleted excessive logging"

This reverts commit d78a46d958cffeabf9543cb0e35e744c7e141bd6.

Revert "SystemLog: execute prepareTable()  on each flush"

This reverts commit 7f60bb0a9b62c991510457efac3386dc573baafb.

SystemLog: execute prepareTable() on each flush

kshvakov committed:
Merge branch 'system-log-prepare-table-on-each-flush' of https://github.com/ClickHouse-Ninja/ClickHouse-Server into system-log-prepare-table-on-each-flush

alexey-milovidov committed:
Update SystemLog.h
